core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/ComponentArchitectureBlankRefreshExtension.java

```diff
@@ -84,6 +89,20 @@ public void beforeRefresh(DDiagram diagram) {
 
     updateTargetDiagram(diagram, !contextualElements.isEmpty());
 
+    // -------------------------------------
+    // Remove all functions added directly on diagram.
+    // Fix for bug 560092, can be deleted in Capella 6.0
+    // -------------------------------------
+
+    String mappingName = MappingConstantsHelper.getMappingFunction(diagram);
+    NodeMapping mapping = DiagramServices.getDiagramServices().getNodeMapping(diagram, mappingName);

```

Comment:
`mapping` can be null, and so line 102 (`mapping.equals(d.getDiagramElementMapping()`) may fail with an NPE.
A null check `if (mapping != null)` is required here before line 99